minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/configuration/ConfigurationChangeCoordinator.java

```diff
@@ -0,0 +1,122 @@
+    private final Set<ConfigurationChangeListener> configurationChangeListeners = new HashSet<>();
+    private final Set<ChangeIngestor> changeIngestors = new HashSet<>();
+
+    /**
+     * Provides an opportunity for the implementation to perform configuration and initialization based on properties received from the bootstrapping configuration
+     *
+     * @param properties from the bootstrap configuration
+     */
+    public void initialize(Properties properties, ConfigurationFileHolder configurationFileHolder) {
+        final String ingestorsCsv = properties.getProperty(NOTIFIER_INGESTORS_KEY);
+
+        if (ingestorsCsv != null && !ingestorsCsv.isEmpty()) {
+            for (String ingestorClassname : Arrays.asList(ingestorsCsv.split(","))) {
+                ingestorClassname = ingestorClassname.trim();
+                try {
+                    Class<?> ingestorClass = Class.forName(ingestorClassname);
+                    ChangeIngestor changeIngestor = (ChangeIngestor) ingestorClass.newInstance();
+                    changeIngestor.initialize(properties, configurationFileHolder, this);
+                    changeIngestors.add(changeIngestor);
+                    logger.info("Initialized ");
+                } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+                    throw new RuntimeException("Issue instantiating ingestor " + ingestorClassname, e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Begins the associated notification service provided by the given implementation.  In most implementations, no action will occur until this method is invoked.
+     */
+    public void start() {
+        changeIngestors.forEach(ChangeIngestor::start);
+    }
+
+    /**
+     * Provides an immutable collection of listeners for the notifier instance
+     *
+     * @return a collection of those listeners registered for notifications
+     */
+    public Set<ConfigurationChangeListener> getChangeListeners() {
+        return Collections.unmodifiableSet(configurationChangeListeners);
+    }
+
+    /**
+     * Adds a listener to be notified of configuration changes
+     *
+     * @param listener to be added to the collection
+     * @return true if the listener was added; false if already registered
+     */
+    public boolean registerListener(ConfigurationChangeListener listener) {
+        return this.configurationChangeListeners.add(listener);
+    }
+
+    /**
+     * Provide the mechanism by which listeners are notified
+     */
+    public Collection<ListenerHandleResult> notifyListeners(ByteBuffer newConfig) {
+        logger.info("Notifying Listeners of a change");
+
+        Collection<ListenerHandleResult> listenerHandleResults = new ArrayList<>(configurationChangeListeners.size());
+        for (final ConfigurationChangeListener listener : getChangeListeners()) {
```

Comment:
In the current implementation we only have one listener (RunMiNiFi) and we call init->register->start all in RunMiNiFi. When we do have more I would imagine we would know exactly what listeners to register when we init. 

So I think what we should do is remove the "registerListener" method and instead make "init" take a list of Listeners too. Thoughts?